vmngclient/session.py

```diff
@@ -19,9 +21,73 @@
 logger = logging.getLogger(get_logger_name(__name__))
 
 
+class SessionNotCreatedError(Exception):
+    pass
+
+
+def assert_admin_role(session: Session, expected_admin: bool) -> None:
+    role = cast(dict, session.get_json("/dataservice/admin/user/role"))
+    if not role["isAdmin"] is expected_admin:
+        raise SessionNotCreatedError("Got different role than expected")
+
+
+def create_session(
+    url: str,
+    port: int,
+    username: str,
+    password: str,
+    admin: bool = True,
+    subdomain: str = None,
+    timeout: int = 30,
+) -> Session:
+    """Factory function that creates session object based on provided arguments.
+
+    Args:
+        url (str): IP address or domain name
+        port (int): port
+        username (str): username
+        password (str): password
+        admin (bool): specifies the type of session that is returned,
+            for 'True' it is 'ProviderSession', if 'False' it is 'TenantSession'
+        subdomain: subdomain specifying to which view switch when creating provider as a tenant session,
+            if provided creates
+        timeout (int): timeout
+
+    Returns:
+        Session object
+    """
+
+    if admin and not subdomain:
+        provider_session = ProviderSession(url, port, username, password, timeout)
+        assert_admin_role(provider_session, expected_admin=True)
+        return provider_session
+    if admin and subdomain:
+        provider_as_a_session = ProviderAsTenantSession(url, port, username, password, subdomain, timeout)
+        assert_admin_role(provider_as_a_session, expected_admin=True)
+        return provider_as_a_session
+    if not admin and not subdomain:
+        tenant_session = TenantSession(url, port, username, password, timeout)
+        assert_admin_role(tenant_session, expected_admin=False)
+        return tenant_session
+
+    raise SessionNotCreatedError("Session not created. Check admin and subdomain.")
```

Comment:
We need to improve exception message. It should give detailed information about exception. In perfect world it should also provide the root cause and solution  

Session to {url}:{port} couldn't established for {user}  due to ....